### DISCUSSION TITLE
ROCm CPU runnable code - first try

### DISCUSSION THREAD

**Discussion by rudifus-ppy** [karpathy/autoresearch#197] [General]
uv run train.py
Using device: cpu
Vocab size: 8,192
Model config: {'sequence_len': 2048, 'vocab_size': 8192, 'n_layer': 8, 'n_head': 4, 'n_kv_head': 4, 'n_embd': 512, 'window_pattern': 'SSSL'}
Parameter counts:
  wte                     : 4,194,304
  value_embeds            : 16,777,216
  lm_head                 : 4,194,304
  transformer_matrices    : 25,166,336
  scalars                 : 16
  total                   : 50,332,176
Estimated FLOPs per token: 2.390784e+08
Scaling AdamW LRs by 1/sqrt(512/768) = 1.224745
Skipping torch.compile on CPU
Time budget: 300s
Gradient accumulation steps: 2

**Comment by JamesStallings** [comment]
I am very interested to know how this goes for you. I have a pretty good footprint in this hardware configuration and am eager to see what sort of outcomes you can get, with an eye to some exploration of my own.
